did_doc_sov/tests/serde_legacy.rs

```diff
@@ -0,0 +1,74 @@
+use did_doc_sov::{extra_fields::KeyKind, DidDocumentSov};
+
+const LEGACY_DID_DOC_JSON: &str = r##"
+{
+   "@context": "https://w3id.org/did/v1",
+   "id": "2ZHFFhzA2XtTD6hJqzL7ux",
+   "publicKey": [
+       {
+           "id": "1",
+           "type": "Ed25519VerificationKey2018",
+           "controller": "2ZHFFhzA2XtTD6hJqzL7ux",
+           "publicKeyBase58": "rCw3x5h1jS6gPo7rRrt3EYbXXe5nNjnGbdf1jAwUxuj"
+       }
+   ],
+   "authentication": [
+       {
+           "type": "Ed25519SignatureAuthentication2018",
+           "publicKey": "2ZHFFhzA2XtTD6hJqzL7ux#1"
+       }
+   ],
+   "service": [
+       {
+           "id": "did:example:123456789abcdefghi;indy",
+           "type": "IndyAgent",
+           "priority": 0,
+           "recipientKeys": [
+               "2ZHFFhzA2XtTD6hJqzL7ux#1"
+           ],
+           "routingKeys": [
+               "8Ps2WosJ9AV1eXPoJKsEJdM3NchPhSyS8qFt6LQUTKv2",
+               "Hezce2UWMZ3wUhVkh2LfKSs8nDzWwzs2Win7EzNN3YaR"
+           ],
+           "serviceEndpoint": "http://localhost:8080/agency/msg"
+       }
+   ]
+}
+"##;
+
+const VERKEY_BASE58: &str = "6MkfJTyeCL8MGvZntdpXzpitL6bM6uwCFz8xcYar18xQBh7";
+
+const DID_PEER: &str = "did:peer:2.Vz6MkfJTyeCL8MGvZntdpXzpitL6bM6uwCFz8xcYar18xQBh7.SeyJwcmlvcml0eSI6MCwiciI6WyI4UHMyV29zSjlBVjFlWFBvSktzRUpkTTNOY2hQaFN5UzhxRnQ2TFFVVEt2MiIsIkhlemNlMlVXTVozd1VoVmtoMkxmS1NzOG5Eeld3enMyV2luN0V6Tk4zWWFSIl0sInJlY2lwaWVudEtleXMiOlsiMlpIRkZoekEyWHRURDZoSnF6TDd1eCMxIl0sInMiOiJodHRwOi8vbG9jYWxob3N0OjgwODAvYWdlbmN5L21zZyIsInQiOiJJbmR5QWdlbnQifQ";
+
+#[test]
+fn test_deserialization_legacy() {
+    let did_doc: DidDocumentSov = serde_json::from_str(LEGACY_DID_DOC_JSON).unwrap();
+    println!("{:#?}", serde_json::to_string_pretty(&did_doc).unwrap());
+    assert_eq!(did_doc.id().to_string(), DID_PEER);
+    assert_eq!(did_doc.verification_method().len(), 1);
+    assert_eq!(did_doc.authentication().len(), 0);
```

Comment:
it seems that the legacy ddo's `publicKey`  get mapped to `verification_method` but legacy ddo's `authentication` gets mapped to nothing. 

Any comments on this? 

Afaik in legacy DDO `public_key` was predecessor of `authentication` which never got widely adopted, so I think this should be alright? I think in the legacy ddo paradigm, everyone in aries was using `public_key` only.

Reply:
The behavior is consistent with deriving the peer did using [this](https://github.com/TimoGlastra/legacy-did-transformation) algorithm and resolving the resulting did (as per the did peer [spec](https://identity.foundation/peer-did-method-spec)).